### PR TITLE
iOS marker selection

### PR DESCRIPTION
**Pull request by karimnaaji**
Marker selection for iOS.

To go in #1161

### CODE REVIEW

File: ios/demo/src/MapViewController.m [hunk 1]
```diff
@@ -32,6 +32,22 @@ - (void)mapView:(TGMapViewController *)mapView didLoadSceneAsync:(NSString *)sce
     [mapView setPosition:newYork];
 }
 
+- (void)mapView:(TGMapViewController *)mapView didSelectMarker:(TGMapMarkerId)markerId atScreenPosition:(CGPoint)position
+{
+    if (markerId == 0) {
```
Comment (msmollin) [inline ios/demo/src/MapViewController.m]: Just curious - would we ever want to call the delegate callback if the markerId is zero? Seems like something I wouldn't expect as an implementing developer.

File: ios/src/TGMapViewController.mm [hunk 2]
```diff
@@ -149,6 +149,32 @@ - (void)pickFeaturesAt:(CGPoint)screenPosition
...
+    if (!self.map && !self.mapViewDelegate) { return; }
+
+    screenPosition.x *= self.contentScaleFactor;
+    screenPosition.y *= self.contentScaleFactor;
+
+    self.map->pickMarkerAt(screenPosition.x, screenPosition.y, [screenPosition, self](const Tangram::MarkerPickResult* markerPickResult) {
+        CGPoint position = CGPointMake(0.0, 0.0);
+
+        if (!markerPickResult) {
+            if ([self.mapViewDelegate respondsToSelector:@selector(mapView:didSelectMarker:atScreenPosition:)]) {
```
Comment (msmollin) [inline ios/src/TGMapViewController.mm]: Oh I see. Hmmm. What value does this add calling with a marker value of zero?

**Comment by msmollin** [comment]
Does this need a rebase to get the circle builds working?

File: ios/demo/src/MapViewController.m (hunk 1 above)
Comment (karimnaaji) [inline ios/demo/src/MapViewController.m]: We want to always have a callback, it's a nicer behavior. As the implementing developer querying the system without knowing if the query has ever completed is misleading; you actually don't know if it actually performed the query at all. So you never really know if it failed for other reasons than not finding a pick result.

**Comment by karimnaaji** [comment]
Yes it does, in the process of rebasing the whole thing :)

**Pull request closed without merging**